programs/bpf_loader/src/syscalls/mem_ops.rs

```diff
@@ -294,15 +294,45 @@ fn memset_non_contiguous(
     accounts: &[SerializedAccountMetadata],
     memory_mapping: &MemoryMapping,
 ) -> Result<u64, Error> {
-    let dst_chunk_iter =
-        MemoryChunkIterator::new(memory_mapping, accounts, AccessType::Store, dst_addr, n)?;
-    for item in dst_chunk_iter {
-        let (dst_region, dst_vm_addr, dst_len) = item?;
-        let dst_host_addr = Result::from(dst_region.vm_to_host(dst_vm_addr, dst_len as u64))?;
-        unsafe { slice::from_raw_parts_mut(dst_host_addr as *mut u8, dst_len).fill(c) }
+    iter_memory_chunks(
+        AccessType::Store,
+        dst_addr,
+        n,
+        accounts,
+        memory_mapping,
+        |slice| {
+            slice.fill(c);
+            Ok(0)
+        },
+    )
+}
+
+pub fn iter_memory_chunks<T, F>(
+    access: AccessType,
+    addr: u64,
+    n_bytes: u64,
+    accounts: &[SerializedAccountMetadata],
+    memory_mapping: &MemoryMapping,
+    mut fun: F,
```

Comment:
This could be a `&mut dyn FnMut`, instead of a generic.

Reply:
Do we really want `&mut dyn FnMut`? 
- It's such a short function, code bloat shouldn't be an issue (I think)
- `fn iter_memory_pair_chunks()` is also generic (so then both should be changed)

generic is "better inlining and optimization, but larger code" and `dyn` is "shorter slower code, indirect calling is expensive and fewer optimizations possible". 

wdty?